packages/block-library/src/avatar/hooks.js

```diff
@@ -76,14 +76,23 @@ export function useUserAvatar( { userId, postId, postType } ) {
 		},
 		[ postType, postId, userId ]
 	);
+	const defaultAvatar = useDefaultAvatar();
+	if ( authorDetails && ! authorDetails?.avatar_urls ) {
+		return {
```

Comment:
If we do a return if this condition applies, we may skip some checks we make after, assuming that sizes or avatar URLs may not be null on the code after this return.

Reply:
yes, great findings.

Reply:
Good Job @manzurahammed @c4rl0sbr4v0

Reply:
Agreed with @c4rl0sbr4v0 
Good Job @manzurahammed

Reply:
@c4rl0sbr4v0 thanks for the tip 😁  done in https://github.com/WordPress/gutenberg/pull/41354/commits/579698f5c4aaadaeeb1e35eebc59f812c9c99312

Reply:
Let me know if you need any help here 😄

Reply:
@c4rl0sbr4v0 please discuss it with other reviewers and make it resolved, thank you for your effort.

Reply:
thanks, @c4rl0sbr4v0 for your information. will inform you after doing this.

Reply:
👋 Hey @rudlinkon, any updates on this? 😄 I'd like to get this merged soon so we can include it in WP 6.0.1.

If you don't find the time currently to apply the [changes](https://github.com/WordPress/gutenberg/pull/41354#discussion_r884657770) that @c4rl0sbr4v0 suggested, that's fine -- just let us know and we'll take care of them 😄

Reply:
@ockham thank you for your reminder, actually, I'm a little bit busy now. @manzurahammed bother could you have a look?

Reply:
Hi, @ockham Could you check, please.
I have pushed a new commit

Reply:
Thanks @rudlinkon and @manzurahammed!

Looking pretty good! I just rebased to fix unit tests which were failing due to an outdated dependency. Let's see if they pass now 🤞